src/training/stylegan.py

```diff
@@ -41,39 +41,44 @@ def __init__(self,
         self.total_epochs = total_epochs
         self.logger = logger
         self.storage = storage
-        self.ocr_loss = ocr_loss.to(device)
-        self.perceptual_loss = perceptual_loss.to(device)
-        self.typeface_loss = typeface_loss.to(device)
         self.ocr_coef = ocr_coef
+        self.cycle_coef = cycle_coef
+        self.recon_coef = recon_coef
         self.emb_coef = emb_coef
         self.perc_coef = perc_coef
         self.tex_coef = tex_coef
-        self.coef_ocr = ocr_coef
-        self.coef_cycle = coef_cycle_loss
-        self.coef_recon = coef_reconstruction_loss
+        self.typeface_loss = typeface_loss.to(device)
+        self.ocr_loss = ocr_loss.to(device)
+        self.perc_loss = perc_loss.to(device)
         self.style_embedder = style_embedder
         self.content_embedder = content_embedder
-        self.l1 = torch.nn.L1Loss() 
+        self.l1 = torch.nn.L1Loss()
 
     def train(self):
         logger.info('Start training')
         self.model.train()
         self.content_embedder.train()
         self.style_embedder.train()
 
-        for style_batch, content_batch, label_batch, style_label_batch in self.train_dataloader:
-            if max(len(label) for label in label_batch) > 25:
+        # style_batch - images containing the style we want to imitate
+        # conent_batch - rendered images containing the conent we want to draw
+        # label_batch - text labels of content batch
+        # style_label_batch - text labels of style batch
+        for style_batch, content_batch, content_label_batch, style_label_batch in self.train_dataloader:
+            if max(len(label) for label in content_label_batch) > 25:
                 continue
 
+            self.optimizer.zero_grad()
+
             style_batch = style_batch.to(self.device)
             content_batch = content_batch.to(self.device)
             style_label_batch = style_label_batch.to(self.device)
+
             style_embeds = self.style_embedder(style_batch)
             content_embeds = self.content_embedder(content_batch)
-            self.optimizer.zero_grad()
 
             res = self.model(content_embeds, style_embeds)
-            ocr_loss, recognized = self.ocr_loss(res, label_batch, return_recognized=True)
+            ocr_loss, recognized = self.ocr_loss(res, content_label_batch, return_recognized=True)
             words = []
             for word in recognized:
```

Comment:
I was thinking about this part of code where we collect recognised words and maybe we should make it a function inside of OCR (but I am not sure).

Reply:
We could move drawing the words inside util/